IFC_Adapter/Methods/Convert/Physical/FromIFC/ReinforcingBar.cs

```diff
@@ -43,10 +44,19 @@ public static IReinforcingBar ReinforcingBarFromIfc(this IIfcReinforcingBar elem
                 return null;
             }
 
+            IfcObject ifcObject = element as IfcObject;
+            if (ifcObject == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("The IFC element could not be read due to an internal error.");
```

Comment:
I'd change the error text to something like
```suggestion
                BH.Engine.Reflection.Compute.RecordError($"The provided IFC element of type {element.GetType().Name} could not be converted to {nameof(IfcObject)}.");
```